text/0000-compiler-team-trusted-contrib-maintainer.md

```diff
@@ -0,0 +1,381 @@
+
+Unlike trusted contributors, team members are expected to consider themselves
+as *maintainers* of the compiler - put otherwise, to be invested in the quality
+of the compiler codebase and overall health of the compiler team, independent
+of their own projects. Team membership is primarily intended to recognise and
+encourage participation in activities which are vital to the success of the
+compiler team and broader project.
+
+Team members are expected to participate in the ongoing maintainance tasks
+that the compiler team is responsible for (with all of the expected caveats
+for vacation time, mental health breaks, etc) - listed as [responsibilities]
+[responsibilities] below. However, not all contributors need to participate
+in these responsibilities to an equal degree. Contributors should participate
+in these tasks to the degree that they are able - volunteers are not expected
+to participate as much as those employed to work on the compiler, for example.
+It is the responsibility of the compiler team leads to ensure that the ongoing
+maintenance tasks of the team can be completed sustainably.
+
+Team members have all of the same permissions and access as trusted
+contributors. Like trusted contributors, team members are considered members of
+the Rust project as a whole and are expected to follow the spirit of the [Code
+of Conduct][coc].
+
+Like trusted contributors, after inactivity for longer than a year, a
+contributor will be moved to alumni status. Members who are no longer able to
+help maintain the compiler but otherwise wish to continue contributing to the
+compiler can also be moved to alumni status and retain their trusted contributor
+status. Alumni can ask to be reinstated in future.
+
+[coc]: https://www.rust-lang.org/policies/code-of-conduct
+
+Responsibilities
+----------------
+[responsibilities]: #responsibilities
+
+There are various responsibilities that a team member could take on to help the
+team. All team members must participate in at least one activity.
+
+Team members can get involved in any of these by contacting the team leads, by
+asking team members currently involved in these responsibilities, or by asking
+in any venue where these responsibilities are conducted (e.g. a Zulip stream).
+
+- Final comment period (FCP) reviewer
+    - Final comment periods are the process by which the team signs-off on a
+      change before it is made, like stabilizing a feature.
+
+      FCPs have always required whole team to sign-off, but this doesn't scale
+      as the team grows. As described above, this acts as a disincentive for the
+      team to grow. Furthermore, not all FCPs are relevant to all team members and a
+      diffusion of responsibility means that most team members just sanity-check and
+      then sign-off. This isn't ideal, as it doesn't guarantee that someone on the
+      team has thoroughly considered a FCP.
+
+      Instead, have FCPs require sign-off from team members who opt-in to being
```

Comment:
I'll note the @rfcbot already supports opt-*out* of being on FCPs. This is [undocumented](https://github.com/rust-lang/team/pull/1419), no one currently uses this and probably not a lot of people know it's an option at all (Mara told me and Boxy about it a couple days ago and both me and Boxy were in complete shock).

But the option does exist and should even work. And it seems that opt-out-ness is deliberate (so it's harder to exclude someone accidentally)...

Reply:
I agree